docs/lang/articles/advanced/differentiable_programming.md

```diff
@@ -169,6 +169,10 @@ for examples on using autodiff-based force evaluation MPM and FEM.
 As mentioned above, `ti.Tape()` can only track a 0D field as the output variable.
```

Comment:
shall we use a few sentences to describe what is differential programming and how it works at beginning of the doc?
Just mentioning it's useful in scientific computations and AI does not offer much information about what it is.